Sources/ClientRuntime/EventStream/DefaultMessageEncoderStream.swift

```diff
@@ -13,39 +13,44 @@ extension EventStream {
         let messageSinger: MessageSigner
         let requestEncoder: RequestEncoder
         var readAsyncIterator: AsyncIterator?
-
-        public init(stream: AsyncThrowingStream<Event, Error>,
-                    messageEncoder: MessageEncoder,
-                    requestEncoder: RequestEncoder,
-                    messageSinger: MessageSigner) {
+        
+        public init(
+            stream: AsyncThrowingStream<Event, Error>,
+            messageEncoder: MessageEncoder,
+            requestEncoder: RequestEncoder,
+            messageSinger: MessageSigner
+        ) {
             self.stream = stream
             self.messageEncoder = messageEncoder
             self.messageSinger = messageSinger
             self.requestEncoder = requestEncoder
             self.readAsyncIterator = makeAsyncIterator()
         }
-
+        
         public struct AsyncIterator: AsyncIteratorProtocol {
             let stream: AsyncThrowingStream<Event, Error>
             let messageEncoder: MessageEncoder
             var messageSinger: MessageSigner
             let requestEncoder: RequestEncoder
-
+            
             private var lastMessageSent: Bool = false
-
-            init(stream: AsyncThrowingStream<Event, Error>,
-                 messageEncoder: MessageEncoder,
-                 requestEncoder: RequestEncoder,
-                 messageSinger: MessageSigner) {
+            private var streamIterator: AsyncThrowingStream<Event, Error>.Iterator
+            
+            init(
+                stream: AsyncThrowingStream<Event, Error>,
+                messageEncoder: MessageEncoder,
+                requestEncoder: RequestEncoder,
+                messageSinger: MessageSigner
+            ) {
                 self.stream = stream
+                self.streamIterator = stream.makeAsyncIterator()
```

Comment:
Previously, we were creating a new iterator everytime `next()` is called. Now we create an iterator once during initi.